fact-ebpf/src/bpf/bound_path.h

```diff
@@ -21,10 +22,13 @@ __always_inline static void path_write_char(char* p, unsigned int offset, char c
   *path_safe_access(p, offset) = c;
 }
 
-__always_inline static struct bound_path_t* path_read(struct path* path) {
+__always_inline static struct bound_path_t* _path_read(struct path* path, bool use_bpf_d_path) {
   struct bound_path_t* bound_path = get_bound_path();
+  if (bound_path == NULL) {
+    return NULL;
+  }
 
-  bound_path->len = bpf_d_path(path, bound_path->path, PATH_MAX);
+  bound_path->len = use_bpf_d_path ? bpf_d_path(path, bound_path->path, PATH_MAX) : d_path(path, bound_path->path, PATH_MAX);
```

Comment:
TIL that bpf_d_path is already [legacy](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d08e2045ebf0f5f2a97ad22cc7dae398b35354ba) :) But for a good reason, looks like there is a possibility of concurrent modification of the path structure, leading to use-after-free -- I assume the same is correct for using copied `d_path` function as well.